x-pack/plugins/spaces/server/lib/create_spaces_service.ts

```diff
@@ -26,6 +27,10 @@ export function createSpacesService(server: any): SpacesService {
     return spaceId;
   }
 
+  function setSpaceId(request: any, spaceId: string) {
```

Comment:
I can remove this, as it's not used yet, but this is a potential solution for allowing Spaces to dictate which space id should be considered the "current" space id for the purpose of authorization.

This has the benefit of not modifying the base path for the in-flight request, so `request.getBasePath()` / `http.getBasePathFor(request)` will not be impacted by calls to `setSpaceId`, which reduces the likelihood of side-effects as a result of changing the space id mid-request.

Reply:
Hmm, I always figured this would just be done for "copy to space" by allowing "consumers" to specify the `namespace` in the options. Otherwise, we can't use the SOC instances for different spaces concurrently and we have do so serially.

Reply:
> Hmm, I always figured this would just be done for "copy to space" by allowing "consumers" to specify the namespace in the options. Otherwise, we can't use the SOC instances for different spaces concurrently and we have do so serially.

Yep, that's the other route we can take! I started down that path in this PR, but ended up [reverting the required changes](https://github.com/elastic/kibana/pull/38444/commits/c98c1cd6a37f03183f1b60d80bf3100d85f70caf#diff-a5ae7917e5d0aa4676dd008d3628d987) for now.

I'm happy to take either approach, I just wanted to present options. I'll revert this change in this PR, and re-introduce the ability for the spaces wrapper to accept a namespace from consumers.